src/interpreter.ts

```diff
@@ -25,3 +499,840 @@ export function parseAndEvalExpression(sourceCode: string): EvalResult {
+                break;
+            case "statement_expression":
+                this.interpretExpressionStatement(ast);
+                break;
+            case "statement_foreach":
+                this.interpretForEachStatement(ast);
+                break;
+            case "statement_repeat":
+                this.interpretRepeatStatement(ast);
+                break;
+            case "statement_return":
+                this.interpretReturnStatement(ast);
+                break;
+            case "statement_try":
+                this.interpretTryStatement(ast);
+                break;
+            case "statement_try_catch":
+                this.interpretTryCatchStatement(ast);
+                break;
+            case "statement_until":
+                this.interpretUntilStatement(ast);
+                break;
+            case "statement_while":
+                this.interpretWhileStatement(ast);
+                break;
+        }
+    }
+
+    public interpretLetStatement(ast: AstStatementLet) {
+        const val = this.interpretExpression(ast.expression);
+        this.envStack.setNewBinding(idText(ast.name), val);
+    }
+
+    public interpretAssignStatement(ast: AstStatementAssign) {
+        const path = ast.path;
+        const exp = ast.expression;
+
+        if (path.kind === "id") {
+            const val = this.interpretExpression(exp);
+            this.envStack.updateBinding(idText(path), val);
+        } else {
+            throwNonFatalErrorConstEval(
+                `only identifiers are currently supported as path expressions`,
+                path.loc,
+            );
+        }
+    }
+
+    public interpretAugmentedAssignStatement(ast: AstStatementAugmentedAssign) {
+        const path = ast.path;
+        const exp = ast.expression;
+        const op = ast.op;
+        const source = ast.loc;
+
+        if (path.kind === "id") {
+            const updateVal = this.interpretExpression(exp);
+            const currentPathValue = this.envStack.getBinding(idText(path));
+            if (currentPathValue === undefined) {
+                throwNonFatalErrorConstEval(`undeclared identifier`, path.loc);
+            }
+            const newVal = evalBinaryOp(
+                op,
+                currentPathValue,
+                updateVal,
+                path.loc,
+                exp.loc,
+                source,
+            );
+            this.envStack.updateBinding(idText(path), newVal);
+        } else {
+            throwNonFatalErrorConstEval(
+                `only identifiers are currently supported as path expressions`,
+                path.loc,
+            );
+        }
+    }
+
+    public interpretConditionStatement(ast: AstCondition) {
+        const condition = ensureBoolean(
+            this.interpretExpression(ast.condition),
+            ast.condition.loc,
+        );
+        if (condition) {
+            this.envStack.executeInNewEnvironment(() => {
```

Comment:
why do you need a new environment here? conditionals in Tact do not introduce new binders

Reply:
ah, nevermind, this is to get rid of local vars from the new statement blocks

Reply:
I think we could probably define a helper function which could be reused for every block statement execution, basically abstracting this
```
this.envStack.executeInNewEnvironment(() => {
                ast.trueStatements.forEach(this.interpretStatement, this);
            });
```

and then reusing it for the else branch, while, repeat, do loops (and the currently missing block statement)

Reply:
or maybe for loops it's not a good idea